IntegrationTests/Tests/IntegrationTests/SwiftPMTests.swift

```diff
@@ -1,106 +1,131 @@
+                    //   '...supported extensions are: zip, tar.gz, tar'
+                    //   '...supported extensions are: tar.gz, zip, tar'
+                    // Only check for the start of that string.
+                    stderr.contains("error: unexpected file type; supported extensions are:")
+                )
 
                 let validPath = fixturePath.appending(component: "SwiftFramework.zip")
-                let (stdout, _) = try sh(swiftPackage, "--package-path", fixturePath, "compute-checksum", validPath)
-                XCTAssertEqual(stdout.spm_chomp(), "d1f202b1bfe04dea30b2bc4038f8059dcd75a5a176f1d81fcaedb6d3597d1158")
+                let (stdout, _) = try sh(
+                    swiftPackage, "--package-path", fixturePath, "compute-checksum", validPath
+                )
+                #expect(
+                    stdout.spm_chomp()
+                        == "d1f202b1bfe04dea30b2bc4038f8059dcd75a5a176f1d81fcaedb6d3597d1158"
+                )
             }
         }
     }
 
-    func testSwiftBuild() throws {
-        #if os(Linux)
-        if FileManager.default.contents(atPath: "/etc/system-release").map { String(decoding: $0, as: UTF8.self) == "Amazon Linux release 2 (Karoo)\n" } ?? false {
-            throw XCTSkip("Skipping SwiftBuild testing on Amazon Linux because of platform issues.")
-        }
-        #endif
-
-        // Test SwiftBuildSystem
+    @Test(nil, .skipHostOS(.linux, "Amazon Linux has platform issues"))
+    func packageInitExecutable() throws {
+        // Executable
         do {
             try withTemporaryDirectory { tmpDir in
                 let packagePath = tmpDir.appending(component: "foo")
                 try localFileSystem.createDirectory(packagePath)
                 try sh(swiftPackage, "--package-path", packagePath, "init", "--type", "executable")
                 try sh(swiftBuild, "--package-path", packagePath, "--build-system", "swiftbuild")
-                // SWBINTTODO: Path issues related to swift run of the output from swiftbuild buildsystem
-                //let (stdout, stderr)  = try sh(swiftRun, "--package-path", packagePath, "--build-system", "swiftbuild")
-                //XCTAssertMatch(stdout, .contains("Hello, world!"))
+                // SWBINTTODO: Path issues related to swift test of the output from a swiftbuild buildsystem
+                // let (stdout, stderr) = try sh(
+                //     swiftRun, "--package-path", packagePath, "--build-system", "swiftbuild"
+                // )
+                // #expect(!stderr.contains("error:"))
+                // #expect(stdout.contains("Hello, world!"))
             }
         }
+    }
 
-        #if !os(Windows)
-        // SWBINTTODO: Windows fails to link this library package due to a "lld-link: error: subsystem must be defined" error. See https://github.com/swiftlang/swift-build/issues/310
+    @Test(
+        nil, .skipHostOS(.linux, "Amazon Linux has platform issues"),
```

Comment:
issue (blocking): this will skip the test on all linux host, but the issue is only on Amazon Linux 2.  Can we update the `skipHostOS` trait that optionally accepts an "enum" which we can use skip against a specific distribution?

if the optional is not set, skip on all host.  if the optional is set, skip only on that distribution.